### PR TITLE
Add configs and optional post embedding MLP for ESM encoder

### PR DESCRIPTION
**Pull request by chaitjo**
This PR implements:
- ESM encoder configs
- Optional post embedding MLP after getting ESM embeddings -- this allows the model to merge structural featurisation with ESM features and could be a stronger baseline.
- Optionally allow freezing or finetuning ESM

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 * Fixes typo in wandb run name setting ([#30](https://github.com/a-r-j/ProteinWorkshop/pull/30))
 * Fixes paths for models and datasets when testing instantiation of each module ([#32](https://github.com/a-r-j/ProteinWorkshop/pull/32))
 * Improvements to TFN, MACE and EGNN models and layers, including DiffDock-style intermediate edge feature creation (TFN), dropout, gaussian RBF, mean global pooling ([#38](https://github.com/a-r-j/ProteinWorkshop/pull/38))
+* Implement ESM embedding encoder ([#33](https://github.com/a-r-j/ProteinWorkshop/pull/33), [#41](https://github.com/a-r-j/ProteinWorkshop/pull/33))
```
Comment (a-r-j) [inline CHANGELOG.md:7]: This should be in a new section for `0.2.5`

**Review by a-r-j — APPROVED** [review]
LGTM!

**Comment by amorehead** [comment]
LGTM. Great work adding the post-MLP branch

File: CHANGELOG.md (hunk 1 above)
Comment (chaitjo) [reply CHANGELOG.md]: Done

**Pull request merged**